### PR TITLE
doc: ETCM-8682 Migration strategy for generic versioned Plutus data

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

Initial draft of the migration strategy.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages.
- [ ] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+This document describes steps needed to migrate a running chain to use the new smart contracts
+that introduce versioned Plutus data.
+
+## Context
+
+To make it easier for Partner Chains to keep up with the evolution of the Partner Chains SDK
+while avoiding chain resets and other disruptions, the new Partner Chains Smart Contracts version
+introduces a new layout of on-chain data in the form of *generic versioned Plutus data*. This
+means that every datum attached to UTxO-s produced by the Partner Chains tooling and observed by
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:13]: Cosmetic: technical writers use `UTXOs`, I would follow them

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- **version number**
+
+## Migration steps - Ariadne
+
+At the moment, the only feature actively supported by the Partner Chains SDK which uses Plutus
+data observability is the Ariadne committee selection mechanism, which observes the following
+data:
+- D-Parameter
+- Permissioned candidates list
+- Cardano SPO registrations
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:26]: This guide will be for v1.4, when in v1.3 we introduce observability of token release from reserve to illiquid supply. Will this script be changed as well?

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+The outline of the migration can be summarized as:
+1. New smart contract release
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:29]: Partner-chains tribe "releases" new versions of smart contracts. I'm looking for a better wording here. Isn't "Migration of Cardano data" more appropriate here?

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+1. Node data source upgrade
+1. Transitionary period
+1. Observed addresses change
+
+### Smart contract redeployment
+
+Introducing versioned Plutus data is a breaking change in the smart contracts, which means
+new versions of the smart contracts will need to be used. The governance authority should use the
+new version of Partner Chains Smart Contracts to set up a new D-Parameter and permissioned
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:38]: This operations should happen when SPOs register with the new smart contracts.

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+changes were made. A version of the node with upgraded data source needs to be made available to
+all participants of the network.
+
+### Transitionary period
+
+At this point a transitionary period should be established when the chain would still
+operate using the old smart contract data. During this period:
+
+- all SPOs wishing to continue participating in the chain will need to register themselves
+again using the new smart contracts.
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:56]: How does SPOs observe their new registrations when their nodes are still configured for the old addresses?

File: docs/developer-guides/versioned-plutus-data-migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+- nodes running in the network will need to be updated to the version released in the previous step.
+At the time of upgrade, no changes to observed addresses should be made, as the new data sources 
+are capable of observing both versioned datums and the legacy schema for backwards compatibility.
+
+### Observed addresses change
+
+After the transitionary period has ended and enough of the nodes runnig the chain have been upgraded
```
Comment (LGLO) [inline docs/developer-guides/versioned-plutus-data-migration-guide.md:63]: runnig => running

**Comment by LGLO** [comment]
In my opinion it is not detailed enough to pass this to users.

For me there should be detailed steps like:
* "governance authority" runs "prepare-configuration" wizards using the v1.4 binaries obtaining new chain-config file (make remark that chains-spec is or is-not needed?)
* "governance authority" publishes this file SPOs
* each SPO has to run "register1", "register2", "register3" wizards using the v1.4 binaries and the new config file (Will they be able to see the registration status? Perhaps depends on base-directory, chain-spec, and others)

File: docs/developer-guides/versioned-plutus-data-migration-guide.md (hunk 2 above)
Comment (AmbientTea) [reply docs/developer-guides/versioned-plutus-data-migration-guide.md:26]: No, native tokens is a completely new mechanism so no migration needed. There will be no legacy version, but the initial implementation can support versioning from the start.

**Pull request closed without merging**